### DISCUSSION TITLE
DatePicker - inputFormat based on locale with allowFreeInput

### DISCUSSION THREAD

**Discussion by oskar-gmerek** [mantinedev/mantine#1341] [Q&A] [answered]
Hi there,

I'm trying to use DatePicker in a multilingual project. 

I used inputFormat and works as expected after choosing a date, but I found out that if I use it with allowFreeInput then UX isn't good. 

For example:

If I change inputFormat to `DD/MM/YYYY` and will put a date in the format `DD/MM/YYYY` then a **date isn't recognized** (default `MM/DD/YYYY` is expected when putting date via keyboard)

This is bad UX for some countries as users will always try to put dates in the format like always they do it.

I'm not pretty sure if I understand it good if inputFormat should take care of this behaviour or if there is maybe another place where this should be configured?

If someone has already done something similar I will be grateful for any guidance

Regards,
Oskar

**Comment by rtivital** [comment, marked as answer]
See documentation – https://mantine.dev/dates/date-picker/#allow-free-input
Provide `dateParser` function

**Reply by rogix** [reply]
Not sure if I understand this behavior. As you can see I typed 07231988, should not the result be July 23, 1988? However the date it displays is totally different and nonsense. Is there a way to fix this? Thanks!

![datepicker](https://user-images.githubusercontent.com/12768403/169318963-d3a26066-d979-41df-bf93-d01f868e4918.gif)

**Reply by rtivital** [reply]
Provide your parser if default one does not work correctly for you – https://mantine.dev/dates/date-picker/#allow-free-input

**Reply by rogix** [reply]
Thanks @rtivital. Is it possible to use a mask on the input? For example when the user type: 31121999 the value in the input changes to 31/12/1999. How can one accomplish this? I tried `value` and `dateParser` but didn't work. Am I missing something?

**Reply by rtivital** [reply]
No, masks are not supported
